test/gcs-test.js

```diff
@@ -0,0 +1,172 @@
+
+const key = random.randomBytes(16);
+
+const contents1 = [
+  Buffer.from('Alex', 'ascii'),
+  Buffer.from('Bob', 'ascii'),
+  Buffer.from('Charlie', 'ascii'),
+  Buffer.from('Dick', 'ascii'),
+  Buffer.from('Ed', 'ascii'),
+  Buffer.from('Frank', 'ascii'),
+  Buffer.from('George', 'ascii'),
+  Buffer.from('Harry', 'ascii'),
+  Buffer.from('Ilya', 'ascii'),
+  Buffer.from('John', 'ascii'),
+  Buffer.from('Kevin', 'ascii'),
+  Buffer.from('Larry', 'ascii'),
+  Buffer.from('Michael', 'ascii'),
+  Buffer.from('Nate', 'ascii'),
+  Buffer.from('Owen', 'ascii'),
+  Buffer.from('Paul', 'ascii'),
+  Buffer.from('Quentin', 'ascii')
+];
+
+const contents2 = [
+  Buffer.from('Alice', 'ascii'),
+  Buffer.from('Betty', 'ascii'),
+  Buffer.from('Charmaine', 'ascii'),
+  Buffer.from('Donna', 'ascii'),
+  Buffer.from('Edith', 'ascii'),
+  Buffer.from('Faina', 'ascii'),
+  Buffer.from('Georgia', 'ascii'),
+  Buffer.from('Hannah', 'ascii'),
+  Buffer.from('Ilsbeth', 'ascii'),
+  Buffer.from('Jennifer', 'ascii'),
+  Buffer.from('Kayla', 'ascii'),
+  Buffer.from('Lena', 'ascii'),
+  Buffer.from('Michelle', 'ascii'),
+  Buffer.from('Natalie', 'ascii'),
+  Buffer.from('Ophelia', 'ascii'),
+  Buffer.from('Peggy', 'ascii'),
+  Buffer.from('Queenie', 'ascii')
+];
+
+describe('GCS', function () {
+  let filter1 = null;
+  let filter2 = null;
+  let filter3 = null;
+  const basicFilter = new BasicFilter();
+  const testFilter = new Golomb(25, new U64(786515));
+  let testFilter1 = null;
+  let testFilter2 = null;
+  let testFilter3 = null;
+
+  it('should test GCS filter build', () => {
+    filter1 = basicFilter.fromItems(key, contents1);
+    assert(filter1);
+    testFilter1 = testFilter.fromItems(key, contents1);
+    assert(testFilter1);
+  });
+
+  it('should test GCS filter copy', () => {
+    filter2 = basicFilter.fromBytes(filter1.n, filter1.toBytes());
+    assert(filter2);
+    testFilter2 = testFilter.fromBytes(testFilter1.n, testFilter1.toBytes());
+    assert(testFilter2);
+    filter3 = basicFilter.fromNBytes(filter1.toNBytes());
+    assert(filter3);
+    testFilter3 = testFilter.fromNBytes(testFilter1.toNBytes());
+    assert(testFilter3);
+  });
+
+  it('filters should not be the same', () => {
+    assert.notDeepStrictEqual(filter1, testFilter1);
+    assert.notDeepStrictEqual(filter2, testFilter2);
+    assert.notDeepStrictEqual(filter3, testFilter3);
+    assert.notBufferEqual(filter1.toBytes(), testFilter1.toBytes());
+    assert.notBufferEqual(filter2.toBytes(), testFilter2.toBytes());
+    assert.notBufferEqual(filter3.toBytes(), testFilter3.toBytes());
+    assert.notBufferEqual(filter1.data, testFilter1.data);
+    assert.notBufferEqual(filter2.data, testFilter2.data);
+    assert.notBufferEqual(filter3.data, testFilter3.data);
+  });
+
+  it('should test GCS filter metadata', () => {
+    assert.strictEqual(filter1.n, contents1.length);
+    assert(filter1.P != null);
+    assert(filter1.M != null);
+    assert(filter1.n != null);
```

Comment:
the `!= null` is redundant here, I think `assert(filter1.P)` is enough. But this is just a note for the future, no need to update the PR again